core/src/main/scala/kafka/server/ClientQuotaManager.scala

```diff
@@ -451,6 +450,35 @@ class ClientQuotaManager(private val config: ClientQuotaManagerConfig,
     }
   }
 
+  /**
+   * Helper method to update quotaTypesEnabled which is a bitwise OR combination of the enabled quota types.
+   * For example:
+   *  - If UserQuotaEnabled = 2 and ClientIdQuotaEnabled = 1, then quotaTypesEnabled = 3 (2 | 1 = 3)
+   *  - If UserClientIdQuotaEnabled = 4 and UserQuotaEnabled = 1, then quotaTypesEnabled =  (4 | 1 = 5)
+   *  - If UserClientIdQuotaEnabled = 4 and ClientIdQuotaEnabled = 2, then quotaTypesEnabled = 6 (4 | 2 = 6)
+   *  - If all three are enabled (1 | 2 | 4), then quotaTypesEnabled = 7
+   */
+  private def updateQuotaTypes(): Unit = {
+    quotaTypesEnabled =  if (clientQuotaCallbackPlugin.isDefined) {
```

Comment:
`quotaTypesEnabled` is a volatile variable. This suggests that we can access it anytime. In this method, we update it in a non atomic manner. I wonder if this could have any undesired impact. It would be better to use a local variable and to update `quotaTypesEnabled` at the end.